modules/jupyter-widget/src/initializer.js

```diff
@@ -0,0 +1,50 @@
+import makeTooltip from './widget-tooltip';
+
+export default function createDeckFromDependencies({
+  dependencies,
+  mapboxApiKey,
+  container,
+  jsonInput,
+  tooltip,
+  onComplete,
+  handleClick
+}) {
+  try {
+    // Filter down to the deck.gl classes of interest
+    const {deck, loaders, mapboxgl} = dependencies;
+    const classesDict = {};
+    const classes = Object.keys(deck).filter(
+      x => (x.indexOf('Layer') > 0 || x.indexOf('View') > 0) && x.indexOf('_') !== 0
+    );
+    classes.map(k => (classesDict[k] = deck[k]));
+
+    loaders.registerLoaders([loaders.CSVLoader]);
+
+    const jsonConverter = new deck.JSONConverter({
+      configuration: {
+        classes: classesDict
+      }
+    });
+
+    const props = jsonConverter.convert(jsonInput);
+
+    const getTooltip = makeTooltip(tooltip);
+
+    const deckgl = new deck.DeckGL({
+      ...props,
+      map: mapboxgl,
+      mapboxApiAccessToken: mapboxApiKey,
+      onClick: handleClick,
+      getTooltip,
+      container
+    });
+    if (onComplete) {
```

Comment:
Why does this function need a callback if nothing is async?

Reply:
This function is async in the Jupyter Notebook-based path [here](https://github.com/uber/deck.gl/pull/3638/files/56af168b2723f70efcfb3c73aee790c5b76e5539#diff-75ded98176e79502b67b21279c633dc6R15).

Reply:
You need to handle the callback in the function where the async operation occurs.